packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -75,19 +75,25 @@ const String oldestDocumentedJavaAgpCompatibilityVersion = '4.2';
 // [_settingsAndroidGradlePluginRegExp] to identify the version section.
 const String _versionGroupName = 'version';
 
-// AGP can be defined in build.gradle
+// AGP can be defined in the dependencies block of [build.gradle] or [build.gradle.kts].
+
 // Expected content:
-// "classpath 'com.android.tools.build:gradle:7.3.0'"
+// Groovy DSL - classpath 'com.android.tools.build:gradle:{{agpVersion}}'
+// Kotlin DSL - classpath("com.android.tools.build.gradle:{{agpVersion}}")
```

Comment:
I think this can also be compileOnly https://github.com/flutter/flutter/blob/9480e889c6d65e41b3523c487ae47f25c6529cc4/packages/flutter_tools/gradle/build.gradle.kts#L32

Reply:
Good catch! I updated the regex to include either classpath or compileOnly and it can be easily extended in the future if we discover more variations.